### PR TITLE
[Command] Added finalize() method

### PR DESCRIPTION
**Pull request by michalkurzeja**
[Console] Added finalize() method to Command class, which is executed after Command::execute() call.
This allows to avoid code repetition for actions that should be performed after the command has finished execution.

| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets |  |
| License | MIT |
| Doc PR |  |

### CODE REVIEW

File: src/Symfony/Component/Console/Command/Command.php [hunk 1]
```diff
@@ -253,6 +272,13 @@ public function run(InputInterface $input, OutputInterface $output)
             $statusCode = $this->execute($input, $output);
         }
 
+        $newStatusCode = $this->finalize($input, $output, $statusCode);
+
+        // if finalize() is extended and doesn't return any value then original status code will be used
+        if ($newStatusCode !== null) {
```
Comment (henrikbjorn) [inline src/Symfony/Component/Console/Command/Command.php]: This should be the responsibility of the finalize implementation.

**Comment by stof** [comment]
I don't think this is necessary. In case you have several possible execution paths and some common finalization logic, you can handle the same thing in the existing ``execute`` method by using a private method for the logic before the finalization. In such case, there is no duplication, and it avoids adding an extra method in the API of the Command class.

for me, it is a :-1:

**Comment by unkind** [comment]
May be you would be interested in [console.terminate](http://symfony.com/doc/2.7/components/console/events.html#the-consoleevents-terminate-event).

**Comment by michalkurzeja** [comment]
Hmm, I suppose with this event you can achieve whatever you could do with finalize(). I guess this pull request is of no use then. Thanks for info :)

**Pull request closed without merging**